File: 1364-tuple-with-same-product/1364-tuple-with-same-product.go
func tupleSameProduct(nums []int) int {
    n, ans, freq := len(nums), 0, make(map[int]int)

    for i := 0; i < n; i++ {
        for j := i + 1; j < n; j++ {
            freq[nums[i] * nums[j]]++
        }
    }

    for  _, val := range freq {
        if (val >= 2) {
            ans += (val * (val - 1) * 4)
        }
    }

    return ans;
}

File: 1364-tuple-with-same-product/1364-tuple-with-same-product.cpp
class Solution {
public:
    int tupleSameProduct(vector<int>& nums) {
        int n = nums.size(), ans = 0;
        unordered_map<int, int> freq;

        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                freq[nums[i] * nums[j]]++;
            }
        }

        for (auto &it : freq) {
            if (it.second >= 2) {
                ans += it.second * (it.second - 1) * 4;
            }
        }

        return ans;
    }
};